src/pineko/scale_variations.py

```diff
@@ -6,55 +6,149 @@
+        return -m * beta.beta_qcd((2, 0), nf)
+    elif delta == 2:
+        if which_part == 0:
+            if logpart == 1:
+                return -m * beta.beta_qcd((3, 0), nf)
+            else:
+                return 0.5 * m * (m + 1) * (beta.beta_qcd((2, 0), nf) ** 2)
+        else:
+            return -(m + 1) * beta.beta_qcd((2, 0), nf)
+
+
+def compute_scalefactor(m, nec_order, to_construct_order, nf, kR):
+    """Compute the factor of renormalization scale variation.
 
-    k_R is defined as the log(mu_{R}^{2}/ Q^2).
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    nec_order : tuple(int)
+        tuple of the order for which the sv contribution is asked for
+    to_contruct_order : tuple(int)
+        tuple of the sv order to be constructed
+    nf : int
+        number of active flavors
+    kR : float
+        log of the ratio between renormalization scale and Q
 
-    nf is the number of active flavors.
+    Returns
+    -------
+    float
+        full contribution of ren sv
     """
-    return 1.0
-
+    delta = to_construct_order[0] - m
+    logpart = to_construct_order[2]
+    return (kR**logpart) * ren_sv_contrib(m, delta, logpart, nec_order[0] - m, nf)
 
-def NLO_ren_var(m, k_R, nf):
-    """NLO ren scale variations depending on the first non vanishing order m.
 
-    k_R is defined as the log(mu_{R}^{2}/ Q^2).
+def compute_map_of_necessary_orders(m, delta):
+    """Compute a dictionary with all the necessary order to compute in order to have the full renormalization scale variation.
 
-    nf is the number of active flavors.
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    delta : int
+        difference between asked order and m
+
+    Returns
+    -------
+    dict(tuple(int))
+        description of all the needed orders
     """
-    return -k_R * m * beta.beta_qcd((2, 0), nf)
-
+    nec_orders = {}
+    for delt in range(delta):
+        nec_orders[(m + delta, 0, delt + 1, 0)] = [
+            (m + de, 0, 0, 0) for de in range(delta - delt)
+        ]
+    return nec_orders
+
+
+def create_svonly_grid(grid, order, new_order, scalefactor):
+    """Create a grid containing only the renormalization scale variations at a given order for a grid."""
+    # Retrieve parameters to create new grid
+    bin_limits = [
+        float(bin) for bin in range(grid.raw.bins() + 1)
+    ]  # +1 because I don't know
+    lumi_grid = [pineappl.lumi.LumiEntry(mylum) for mylum in grid.raw.lumi()]
+    subgrid_params = pineappl.subgrid.SubgridParams()
+    new_order = [pineappl.grid.Order(*new_order)]
+    # create new_grid with same lumi and bin_limits of the original grid but with new_order
+    new_grid = pineappl.grid.Grid.create(
+        lumi_grid, new_order, bin_limits, subgrid_params
+    )
+    # extract the relevant order to rescale from the grid for each lumi and bin
+    grid_orders = [orde.as_tuple() for orde in grid.orders()]
```

Comment:
It is fine to shadow in a more internal scope:
```suggestion
    grid_orders = [order.as_tuple() for order in grid.orders()]
```
otherwise you can use `order_`, if you really don't want to